public/app/services/searchService.js

```diff
@@ -11,10 +11,12 @@
                 var queryString;
 
                 queryString = function (parameters) {
-                    return _.map(parameters, function (value, key) {
-                        return '(' + key + ':' + value + ')';
+                    return _.map(parameters.children, function (child) {
+                        return child.hasOwnProperty('children') ?
+                            '(' + queryString(child) + ')' :
+                            '(' + child.field + ':' + child.value + ')'; // TODO Use comparator
```

Comment:
Oops just saw this TODO.  This means that all comparators will be treated as `contains`, rather than what's actually selected.  I'll leave it up to you whether to merge this as-is and raise a bug for this ("comparators do not work correctly in advanced search") or if you want me to change it in this PR.